plugins/es-analysis-common/src/test/java/io/crate/analysis/common/FulltextITest.java

```diff
@@ -137,11 +140,13 @@ public void testMatchNotOnSubColumn() throws Exception {
         execute("select * from matchbox where match(o['m'], 'Ford')");
         assertThat(response.rowCount(), is(1L));
 
-        expectedException.expect(SQLActionException.class);
-        expectedException.expectMessage("Can only use MATCH on columns of type STRING or GEO_SHAPE, not on 'undefined'");
-
-        execute("select * from matchbox where match(o_ignored['a'], 'Ford')");
-        assertThat(response.rowCount(), is(0L));
+        assertThrows(() -> execute("select * from matchbox where match(o_ignored['a'], 'Ford')"),
+                     isSQLError(is("Can only use MATCH on columns of type STRING or GEO_SHAPE, not on 'undefined'"),
+                                INTERNAL_ERROR,
+                                BAD_REQUEST,
+                                4000));
+        // this assert was never executed before since the execption was thrown, the response is not updated.
+        // assertThat(response.rowCount(), is(0L));
```

Comment:
This assert was never executed before since the execption was thrown, the response is not updated.